pkg/alias/hostAliasGenerator.go

```diff
@@ -73,86 +78,73 @@ func (d *hostAliasGenerator) getGeneratorConfig() (*generatorConfig, error) {
+	hostsConfig.additionalHosts, err = d.retrieveAdditionalHosts(globalCfg)
 	if err != nil {
 		return nil, err
 	}
 
 	return hostsConfig, nil
 }
 
-func (d *hostAliasGenerator) getFQDN() (string, error) {
-	fqdn, err := d.globalConfig.Get(fqdnKey)
-	if err != nil {
-		return "", fmt.Errorf("failed to get value for '%s' from global config: %w", fqdnKey, err)
+func (d *HostAliasGenerator) getFQDN(globalCfg config.GlobalConfig) (string, error) {
+	fqdn, ok := globalCfg.Get(fqdnKey)
+	if !ok {
+		return "", fmt.Errorf("key: %s does not exist in global config", fqdnKey)
 	}
 
-	return fqdn, nil
+	return fqdn.String(), nil
 }
 
-func (d *hostAliasGenerator) isInternalIPUsed() (useInternalIP bool, err error) {
-	useInternalIPRaw, err := d.globalConfig.Get(useInternalIPKey)
-	if err != nil && !registry.IsKeyNotFoundError(err) {
-		return false, fmt.Errorf("failed to get value for '%s' from global config: %w", useInternalIPKey, err)
-	} else if err == nil {
-		useInternalIP, err = strconv.ParseBool(useInternalIPRaw)
-		if err != nil {
-			return false, fmt.Errorf("failed to parse value '%s' of field '%s' "+
-				"in global config: %w", useInternalIPRaw, useInternalIPKey, err)
-		}
+func (d *HostAliasGenerator) isInternalIPUsed(globalCfg config.GlobalConfig) (useInternalIP bool, err error) {
+	useInternalIPRaw, ok := globalCfg.Get(useInternalIPKey)
+	if !ok {
+		return false, fmt.Errorf("key: %s does not exist in global config", useInternalIPKey)
+	}
+
+	useInternalIP, err = strconv.ParseBool(useInternalIPRaw.String())
+	if err != nil {
+		return false, fmt.Errorf("failed to parse value '%s' of field '%s' in global config: %w", useInternalIPRaw, useInternalIPKey, err)
 	}
 
 	return useInternalIP, nil
 }
 
-func (d *hostAliasGenerator) getInternalIP() (internalIP net.IP, err error) {
-	internalIPRaw, err := d.globalConfig.Get(internalIPKey)
-	if err != nil && !registry.IsKeyNotFoundError(err) {
-		return nil, fmt.Errorf("failed to get value for field '%s' from global config: %w", internalIPKey, err)
-	} else if err == nil {
-		internalIP, err = parseInternalIP(internalIPRaw)
-		if err != nil {
-			return nil, err
-		}
+func (d *HostAliasGenerator) getInternalIP(globalCfg config.GlobalConfig) (net.IP, error) {
+	internalIPRaw, ok := globalCfg.Get(internalIPKey)
+	if !ok {
+		return nil, fmt.Errorf("key: %s does not exist in global config", internalIPKey)
 	}
 
-	return internalIP, nil
+	ip := net.ParseIP(internalIPRaw.String())
+	if ip == nil {
+		return nil, fmt.Errorf("failed to parse value '%s' of field '%s' in global config: not a valid ip", internalIPRaw, internalIPKey)
+	}
+
+	return ip, nil
 }
 
-func (d *hostAliasGenerator) retrieveAdditionalHosts() (map[string]string, error) {
-	globalConfig, err := d.globalConfig.GetAll()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get all keys from config: %w", err)
-	}
+func (d *HostAliasGenerator) retrieveAdditionalHosts(globalCfg config.GlobalConfig) (map[string]string, error) {
```

Comment:
🚫 **[checkstyle]** <unparam> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>(*HostAliasGenerator).retrieveAdditionalHosts - result 1 (error) is always nil
<!-- __reviewdog__:ChBiNWQzZmNiYzkyMTUzYzZiEgpjaGVja3N0eWxl -->